cms/static/js/views/pages/container.js

```diff
@@ -548,6 +566,85 @@ function($, _, Backbone, gettext, BasePage,
             // Code in 'base.js' normally handles toggling these dropdowns but since this one is
             // not present yet during the domReady event, we have to handle displaying it ourselves.
             subMenu.classList.toggle('is-shown');
+
+            if (!subMenu.classList.contains('is-shown') && this.options.isIframeEmbed) {
+                try {
+                    window.parent.postMessage(
+                        {
+                            type: 'toggleCourseXBlockDropdown',
+                            message: 'Adjust the height of the dropdown menu',
+                            payload: { courseXBlockDropdownHeight: 0 }
+                        }, document.referrer
+                    );
+                } catch (error) {
+                    console.error(error);
+                }
+            }
+
+            // Calculate the viewport height and the dropdown menu height.
+            // Check if the dropdown would overflow beyond the iframe height based on the user's click position.
+            // If the dropdown overflows, adjust its position to display above the click point.
+            const courseUnitXBlockIframeHeight = window.innerHeight;
+            const courseXBlockDropdownHeight = subMenu.offsetHeight;
+            const clickYPosition = event.clientY;
+
+            function sendMessageToParent(type, message, payload) {
+                try {
+                    if (this.options.isIframeEmbed) {
+                        window.parent.postMessage(
+                            {
+                                type,
+                                message,
+                                payload
+                            },
+                            document.referrer
+                        );
+                    }
+                } catch (e) {
+                    console.error(e);
+                }
+            }
+
+            if (courseUnitXBlockIframeHeight < courseXBlockDropdownHeight) {
+                // If the dropdown menu is taller than the iframe, adjust the height of the dropdown menu.
+                sendMessageToParent.call(
```

Comment:
If you define `sendMessageToParent` as an arrow function (`const sendMessageToParent = (type, message, payload) => {`) then you don't need this old-fashioned `.call(this, ...)` binding.

Reply:
I decided to abandon the sendMessageToParent function so as not to add additional logic.
https://github.com/openedx/edx-platform/pull/35694/commits/8d5b7e3bb77cf48352f1e7864636ac05b31810e1#diff-1b7855448b6086c2a313e0b9ecff179e991106aac52648d880f6f9d551e64577L591